### PR TITLE
Handle zero length lists when interpolating

### PR DESCRIPTION
**Pull request by birtles**
These are the Servo side changes for [Gecko bug 1377053](https://bugzilla.mozilla.org/show_bug.cgi?id=1377053).

These patches have been reviewed by @hiikezoe.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17613)

### CODE REVIEW

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -828,6 +828,10 @@ macro_rules! repeated_vec_impl {
         $(impl<T: RepeatableListAnimatable> Animatable for $ty {
             fn add_weighted(&self, other: &Self, self_portion: f64, other_portion: f64)
                 -> Result<Self, ()> {
+                // If the length of either list is zero, the least common multiple is undefined.
+                if cmp::min(self.len(), other.len()) < 1 {
```
Comment (nox) [inline components/style/properties/helpers/animated_properties.mako.rs]: This would have been better written as `self.is_empty() || other.is_empty()`.

Comment (birtles) [inline components/style/properties/helpers/animated_properties.mako.rs]: Ok, I'll make a follow up for that soon.

Comment (nox) [inline components/style/properties/helpers/animated_properties.mako.rs]: Oh I thought this had already landed, that was just a FYI, don't worry about it.

Comment (birtles) [inline components/style/properties/helpers/animated_properties.mako.rs]: Created https://github.com/servo/servo/pull/17683 for this anyway

**Pull request merged**